src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -119,7 +119,9 @@ private void setSubmissionData(Map<String, Object> data, String overseasEntityId
                         new SubmissionNotFoundException(
                                 String.format("Empty submission returned when generating filing for %s", overseasEntityId)));
 
-        data.put(ENTITY_NAME_FIELD, submissionDto.getEntityName());
+        if (Objects.nonNull(submissionDto.getEntityName())) {
```

Comment:
i'm not sure what will happen if we don't add the ENTITY_NAME_FIELD to the map, so I would keep the map structure the same as previous and populate ENTITY_NAME_FIELD with null if it is missing.
eg
if (Objects.isNull(submissionDto.getEntityName()) {
    data.put(ENTITY_NAME_FIELD, null);
} else {
  data.put(ENTITY_NAME_FIELD, submissionDto.getEntityName().getName());
}

Reply:
My feeling on this is that we should not worry about the null check and just blow up with a null pointer at this point, if no name section is present (it absolutely should be by this point). Need a chat probably to discuss why...

Reply:
Update the Filing service to use the if else conditional and pass null when the block is missing.